Sources/SecureXPC/Client/XPCMachClient.swift

```diff
@@ -11,20 +11,8 @@ import Foundation
 ///
 /// In the case of this framework, the Mach service is expected to be represented by an `XPCMachServer`.
 internal class XPCMachClient: XPCClient {
-    
-	/// Creates and returns a connection for the Mach service represented by this client.
-	/// 
-	/// > Note: This client implementation intentionally does not store a reference to the ``xpc_connection_t`` as it can become
-	/// invalid for numerous reasons. Since it's expected relatively few messages will be sent and the lowest possible
-	/// latency isn't needed, it's simpler to always create the connection on demand each time a message is to be sent.
-	internal override func createConnection() -> xpc_connection_t {
-		let connection = xpc_connection_create_mach_service(self.serviceName, nil, 0)
-		xpc_connection_set_event_handler(connection, { (event: xpc_object_t) in
-			// A block *must* be set as the handler, even though this block does nothing.
-			// If it were not set, a crash would occur upon calling xpc_connection_resume.
-		})
-		xpc_connection_resume(connection)
-
-		return connection
-	}
+    /// Creates and returns a connection for the Mach service represented by this client.
+    internal override func createConnection() -> xpc_connection_t {
+        xpc_connection_create_mach_service(self.serviceName!, nil, 0)
```

Comment:
Are you thinking perhaps having out own `machServiceName: String` and then having an `override var serviceName: String? { machServiceName }`?

Reply:
I don't believe there's any need for the super class to have access to `serviceName`. It appears to never be used. So yes to the first part, but no to the second.

Reply:
It would be used in debug descriptions of the object, and xpc endpoints (in the future when I add the ability to export a client as an endpoint)

Reply:
Is there a reason not to take the same approach as `XPCServer` where each subclass actually implements this and the superclass just has:
```swift
public var endpoint: XPCServerEndpoint {
    fatalError("Abstract Method")
}
```

In regards to debug descriptions, I'd think there'd be advantages to having the subclass do that because each subclass has the most comprehensive set of contextual information.

Reply:
> In regards to debug descriptions, I'd think there'd be advantages to having the subclass do that because each subclass has the most comprehensive set of contextual information.

To clarify, I didn't necessarily mean `CustomDebugStringConvertible` (although we should implement that, too), but the breakdown of the value when displayed in the debugger. Stored properties are nice because they're shown there directly, without needing `po obj` or the like.